carmajava/signal_plugin/src/main/java/gov/dot/fhwa/saxton/carma/signal_plugin/ead/trajectorytree/CoarsePathNeighbors.java

```diff
@@ -129,17 +132,13 @@ public void initialize(List<IntersectionData> intersections, int numIntersection
                 speedAtNext = Math.sqrt(curSpeed*curSpeed + 2.0*maxAccel_*Math.max(distToNext-lagDist, 0.0));
                 timeAtNext += (speedAtNext - curSpeed) / maxAccel_;
             }else {
-                // Only continuos accelerations are allowed so we will reduce our max accel in this case to reach top smooth speed at intersection
-                //timeAtNext += (2.0 * (distToNext - lagDist)) / (curSpeed + operSpeed_);
-                // An alternative approach is to accelerate to operating speed before reaching the intersection.
-                // However, this can make the target distance un-achievable if the fine plan time steps do not line up with the inflection point when max speed is achieved. 
+                //accelerate to operating speed before reaching the intersection.
+                //need to use an intermediate node to check conflicts
+                intermediateNode = new Node(curLoc + distToOperSpeed, timeAtNext, speedAtNext);
```

Comment:
The time setting for this should be curTime + timeToOperSpeed

I think around line 153. We need to update the intermediateNode in case we shift the timeAtNext value.

Reply:
Good catch!